### PR TITLE
breaking: Issue-112 Fix Default Nested Protobuf Types

### PR DESCRIPTION
**Pull request by MikeDalby**
Fixes https://github.com/kafkajs/confluent-schema-registry/issues/112

This change accounts for the fact that the protobufjs `Type` class extends the `Namespace` class. When traversing down the Namespaces we should stop on the first message Type instance, not just the first Namespace without an inner/nested Namespace.

### CODE REVIEW

File: src/ProtoSchema.ts [hunk 1]
```diff
@@ -19,7 +19,9 @@ export default class ProtoSchema implements Schema {
     if (!parent) throw new ConfluentSchemaRegistryArgumentError('no nested fields')
     const keys = Object.keys(parent)
     const reflection = parent[keys[0]]
-    if (reflection instanceof Namespace && reflection.nested)
+
+    // Traverse down the nested Namespaces until we find a message Type instance (which extends Namespace)
+    if (reflection instanceof Namespace && !(reflection instanceof Type) && reflection.nested)
```
Comment (MikeDalby) [inline src/ProtoSchema.ts:24]: Open Question: Is it enough to just check that it's not a Type and has a nested object?
```javascript
if(!(reflection instanceof Type) && reflection.nested)
```

**Comment by Nevon** [comment]
Let's test that this works as intended. There's a test file called something like `newapi` that has a few different protobuf schemas, but none with nested message types. You could extend that suite with a test checks that this finds the correct message type and can encode/decode with it.

File: src/ProtoSchema.ts (hunk 1 above)
Comment (haraldkubota) [reply src/ProtoSchema.ts:24]: I tried the patch as per @MikeDalby and it helped me for a nested ProtoBuf. Went from 
```
  decodedValue: MealItems { itemName: 'pizza' }
```
to a much better
```
  decodedValue: Meal {
    item: [],
    drink: [ [DrinkItems], [DrinkItems] ],
    name: 'pizza'
  }
```
Producer was this Python snippet:
```
mybeer = meal_pb2.Meal.DrinkItems(drink_name="beer")
mywine = meal_pb2.Meal.DrinkItems(drink_name="wine")

meal = meal_pb2.Meal(name='pizza', drink=[mybeer,mywine])
# Less meal (for testing kafkajs which seems to miss the drinks
#meal = meal_pb2.Meal(name='pizza', drink=[])

producer.produce(topic=topic, key=str(uuid4()), value=meal)
```
While I'm missing the "beer" and "wine", at least I got drinks now.

**Comment by MikeDalby** [comment]
@Nevon I've added some tests based on the existing ones. Feel free to let me know if there's anything you'd like to see here.

**Comment by Nevon** [comment]
Excellent! Thanks for the contribution.

One thing that would be really helpful is if you could provide a short summary of how this change affects Protobuf users and what they need to do in order to maintain backwards compatibility (I guess explicitly provide a message type?), so I can add that in the release notes.

**Comment by MikeDalby** [comment]
@Nevon How does something like this sound:

Under certain circumstances, when creating an instance of `SchemaRegistry` for Protobuf without the `messageName` parameter confluent-schema-registry would default to the wrong message type in the schema. Specifically, instead of defaulting to the first message type in the schema it would erroneously default to the first message type _that did not define a nested type_.

If you were relying on this behavior may need to either:
* Start passing the `messageName` parameter instead of relying on the default behavior
* Update your Schema's and re-ingest messages accordingly

See Issue https://github.com/kafkajs/confluent-schema-registry/issues/112 for more info

**Comment by MikeDalby** [comment]
@Nevon Just curious what the release process looks like / if there's a release schedule. Thanks!

**Comment by MikeDalby** [comment]
@Nevon Is there anything I can do to help get a released published?

**Comment by Nevon** [comment]
I've just had _a lot_ going on both professionally and personally, so I haven't gotten around to it. Managed to get v3.0.0 out just now. Sorry about the delay.

**Comment by MikeDalby** [comment]
@Nevon I totally understand and sorry for pestering you. Thank you for all of your hard work!

**Pull request merged**